File: Iteration1/SourceCode/src/main/java/Course.java
import java.util.ArrayList;


public class Course {
    private String name;  //name of the course
    private String code; //course code
    private ArrayList<CourseSection>courseSection=new ArrayList<CourseSection>();
    private ArrayList<Course>prerequisite=new ArrayList<Course>(); //list of prerequisites of the course
    private int semester; // to define semester of the course
    private int credit;  //credit of course
    private String type; // TE NTE UE FC DC


    public Course(String courseName, String courseCode,ArrayList<CourseSection> courseSection, ArrayList<Course> coursePrerequisite, int courseSemester,
                  int courseCredit,String courseType){
        setName(courseName);
        setCode(courseCode);
        setPrerequisite(coursePrerequisite);
        setSemester(courseSemester);
        setCredit(courseCredit);
        setType(courseType);
        setCourseSection(courseSection);

    }

    public ArrayList<CourseSection> getCourseSection() {
        return courseSection;
    }

    public void setCourseSection(ArrayList<CourseSection> courseSection) {
        this.courseSection = courseSection;
    }


    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public ArrayList<Course> getPrerequisite() {
        return prerequisite;
    }

    public void setPrerequisite(ArrayList<Course> prerequisite) {
        this.prerequisite = prerequisite;
    }

    public int getSemester() {
        return semester;
    }

    public void setSemester(int semester) {
        this.semester = semester;
    }

    public int getCredit() {
        return credit;
    }

    public void setCredit(int credit) {
        this.credit = credit;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public void checkPrerequisite(Course course){
        System.out.println( course.prerequisite);
    }

}

File: Iteration3/SourceCode/Course.py
class Course(object):


    def __init__(self, courseName, courseCode, courseSection, coursePrerequisite, courseSemester, courseCredit, courseType):
        self.__name = None
        self.__code = None
        self.__prerequisite = []
        self.__semester = 0
        self.__credit = 0
        self.__type = None

        self.setName(courseName)
        self.setCode(courseCode)
        self.setPrerequisite(coursePrerequisite)
        self.setSemester(courseSemester)
        self.setCredit(courseCredit)
        self.setType(courseType)
        self.__courseSection = courseSection

    def getCourseSection(self):
        return self.__courseSection

    def setCourseSection(self, courseSection):
        self.__courseSection = courseSection


    def getName(self):
        return self.__name

    def setName(self, name):
        self.__name = name

    def getCode(self):
        return self.__code

    def setCode(self, code):
        self.__code = code

    def getPrerequisite(self):
        return self.__prerequisite

    def setPrerequisite(self, prerequisite):
        self.__prerequisite = prerequisite

    def getSemester(self):
        return self.__semester

    def setSemester(self, semester):
        self.__semester = semester

    def getCredit(self):
        return self.__credit

    def setCredit(self, credit):
        self.__credit = credit

    def getType(self):
        return self.__type

    def setType(self, type):
        self.__type = type

    def checkPrerequisite(self, course):
        pass
